experiments/process_sandbox/src/libsandbox.cc

```diff
@@ -692,20 +571,44 @@ namespace sandbox
   Library::~Library()
   {
     wait_for_child_exit();
+    {
+      auto [g, pm] = SharedAllocConfig::Pagemap::get_pagemap_writeable();
+      snmalloc::address_t base =
+        snmalloc::address_cast(memory_provider.get_base());
+      auto top = snmalloc::address_cast(memory_provider.top_address());
+      // Scan the pagemap for all memory associated with this and deallocate
+      // the metaslabs.  Note that we don't need to do any cleanup for the
+      // memory referenced by these metaslabs: it will all go away when the
+      // shared memory region is deallocated.
```

Comment:
And similarly for any metaslabs (and their slabs' memory) not in the shared region.